src/ImageSharp/Common/Helpers/Vector128Utilities.cs

```diff
@@ -62,13 +62,25 @@ public static bool SupportsShiftByte
     /// <param name="vector">The input vector from which values are selected.</param>
     /// <param name="control">The shuffle control byte.</param>
     /// <returns>The <see cref="Vector128{Single}"/>.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static Vector128<float> Shuffle(Vector128<float> vector, [ConstantExpected] byte control)
     {
         if (Sse.IsSupported)
         {
             return Sse.Shuffle(vector, vector, control);
         }
 
+        if (AdvSimd.IsSupported)
+        {
+#pragma warning disable CA1857 // A constant is expected for the parameter
+            Vector128<float> result = Vector128.Create(AdvSimd.Extract(vector, (byte)(control & 0x3)));
+            result = AdvSimd.Insert(result, 1, AdvSimd.Extract(vector, (byte)((control >> 2) & 0x3)));
+            result = AdvSimd.Insert(result, 2, AdvSimd.Extract(vector, (byte)((control >> 4) & 0x3)));
+            result = AdvSimd.Insert(result, 3, AdvSimd.Extract(vector, (byte)((control >> 6) & 0x3)));
+#pragma warning restore CA1857 // A constant is expected for the parameter
+            return result;
+        }
```

Comment:
This is going to be a perf killer in most cases.  It'll be worth testing any path that uses this to make sure it doesn't regress to slower than scalar when Arm starts taking a SIMD path it skipped before.

Reply:
I tossed a coin over this one. My thought was I was making enough savings elsewhere in calling method to justify the cost here. I need to see what the codegen is like, but I have no means to do so currently.

Reply:
I assumed codegen would just be the 4 extracts and 3 inserts (which would make the shuffle a minimum of 8 cycles vs the 1 it takes on x86), but it's actually worse.

![image](https://github.com/SixLabors/ImageSharp/assets/7929102/6ef2f62b-fc60-4a99-9363-1bcbedd141ac)

Reply:
🤮I'lll um, i'll remove that then.

Reply:
I wish that you could define constants from a value given a parameter is a constant. I bet the codegen would be much better then. 

```diff
- public static Vector128<float> Shuffle(Vector128<float> vector, [ConstantExpected] byte control)
+ public static Vector128<float> Shuffle(Vector128<float> vector, const byte control)
```

Then...

```c#
const byte w = (byte)((control >> 6) & 0x3)
```

Reply:
I would have thought the fact that it's inlined and the control byte *is* a constant would have been enough, so I'm quite surprised at that codegen.  I'll have a look in the morning and see if I can figure out what's up.

Reply:
Looks to be an issue with where the constant folding happens.

The inlininer is happening and the operands seen are effectively:
```
               [000000] ----------- arg0                    +--*  LCL_VAR   simd16<System.Runtime.Intrinsics.Vector128`1> V00 arg0         
               [000006] ----------- arg1                    +--*  CNS_INT   int    0
               [000007] ----------- arg2                    +--*  LCL_VAR   simd16<System.Runtime.Intrinsics.Vector128`1> V00 arg0         
               [000010] ----------- arg3                    \--*  AND       int   
               [000008] -----------                            +--*  CNS_INT   int    228
               [000009] -----------                            \--*  CNS_INT   int    3
```

We're likely missing an attempt to constant fold `AND` which is blocking it, which is unfortunate.

Reply:
Going to see if its a trivial fix I can get in...

Reply:
https://github.com/dotnet/runtime/pull/97901 fixes this case (and similar scenarios) in particular

Reply:
Hero!